### PR TITLE
Added yieldIfContendedSafely support

### PR DESCRIPTION
**Pull request by DHuckaby**

### CODE REVIEW

File: sqlbrite/src/main/java/com/squareup/sqlbrite/SqlBrite.java [hunk 1]
```diff
@@ -429,6 +429,37 @@ public int update(@NonNull String table, @NonNull ContentValues values,
...
+   * that there are no nested transactions (beginTransaction has only been called once) and will
+   * throw an exception if that is not the case.
+   * @param sleepAfterYieldDelay if > 0, sleep this long before starting a new transaction if
+   *   the lock was actually yielded. This will allow other background threads to make some
+   *   more progress than they would if we started the transaction immediately.
+   * @return true if the transaction was yielded
+   *
+   * @see SQLiteDatabase#yieldIfContendedSafely(long)
+   */
+  public boolean yieldIfContendedSafely(long sleepAfterYieldDelay) {
```
Comment (JakeWharton) [inline sqlbrite/src/main/java/com/squareup/sqlbrite/SqlBrite.java]: This has no units which is an awful API–no surprise from Android. Let's change it to:
```java
public boolean yieldIfContendedSafely(long sleepAmount, TimeUnit sleepUnit)
```

File: sqlbrite/src/main/java/com/squareup/sqlbrite/SqlBrite.java [hunk 2]
```diff
@@ -429,6 +429,37 @@ public int update(@NonNull String table, @NonNull ContentValues values,
...
+   * successful so far. Do not call setTransactionSuccessful before calling this. When this
+   * returns a new transaction will have been created but not marked as successful. This assumes
+   * that there are no nested transactions (beginTransaction has only been called once) and will
+   * throw an exception if that is not the case.
+   * @return true if the transaction was yielded
+   *
+   * @see SQLiteDatabase#yieldIfContendedSafely()
+   */
+  public boolean yieldIfContendedSafely() {
+      return getWriteableDatabase().yieldIfContendedSafely();
```
Comment (JakeWharton) [inline sqlbrite/src/main/java/com/squareup/sqlbrite/SqlBrite.java]: two space indent

File: sqlbrite/src/main/java/com/squareup/sqlbrite/SqlBrite.java [hunk 3]
```diff
@@ -429,6 +429,37 @@ public int update(@NonNull String table, @NonNull ContentValues values,
...
+   * throw an exception if that is not the case.
+   * @param sleepAfterYieldDelay if > 0, sleep this long before starting a new transaction if
+   *   the lock was actually yielded. This will allow other background threads to make some
+   *   more progress than they would if we started the transaction immediately.
+   * @return true if the transaction was yielded
+   *
+   * @see SQLiteDatabase#yieldIfContendedSafely(long)
+   */
+  public boolean yieldIfContendedSafely(long sleepAfterYieldDelay) {
+      return getWriteableDatabase().yieldIfContendedSafely(sleepAfterYieldDelay);
```
Comment (JakeWharton) [inline sqlbrite/src/main/java/com/squareup/sqlbrite/SqlBrite.java]: two space indent

**Comment by JakeWharton** [comment]
Thanks. Now to wait 20 minutes for the CI...

File: sqlbrite/src/main/java/com/squareup/sqlbrite/SqlBrite.java [hunk 4]
```diff
@@ -429,6 +430,37 @@ public int update(@NonNull String table, @NonNull ContentValues values,
...
+   * throw an exception if that is not the case.
+   * @param sleepAmount if > 0, sleep this long before starting a new transaction if
+   *   the lock was actually yielded. This will allow other background threads to make some
+   *   more progress than they would if we started the transaction immediately.
+   * @return true if the transaction was yielded
+   *
+   * @see SQLiteDatabase#yieldIfContendedSafely(long)
+   */
+  public boolean yieldIfContendedSafely(long sleepAmount, TimeUnit sleepUnit) {
+    return getWriteableDatabase().yieldIfContendedSafely(sleepUnit.toMillis(sleepAmount));
```
Comment (DHuckaby) [inline sqlbrite/src/main/java/com/squareup/sqlbrite/SqlBrite.java]: I believe this is the correct syntax, if you have a better solution please let me know.

Comment (JakeWharton) [inline sqlbrite/src/main/java/com/squareup/sqlbrite/SqlBrite.java]: Yep. This is perfect.

**Comment by JakeWharton** [comment]
It looks like you haven't signed our CLA. Can you sign it at http://squ.re/sign-the-cla

**Comment by DHuckaby** [comment]
@JakeWharton Woops, done.

**Comment by JakeWharton** [comment]
Thanks.

**Pull request merged**